src/makernote_int.cpp

```diff
@@ -1151,15 +1154,6 @@ namespace Exiv2 {
         return aix == nullptr ? -1 : aix->idx_;
     }
 
-    int nikonAf2Selector(uint16_t tag, const byte* /*pData*/, uint32_t size, TiffComponent* const /*pRoot*/)
```

Comment:
We cannot simply throw away this code.  The functionality can be included in nikonSelector.

```
    int nikonSelector(uint16_t tag, const byte* pData, uint32_t size, TiffComponent* const /*pRoot*/)
    {
        if (size < 4) return -1;
        const NikonArrayIdx* aix = find(nikonArrayIdx, NikonArrayIdx::Key(tag, reinterpret_cast<const char*>(pData), size));
        return aix == nullptr ? -1 : aix->idx_;
    }
```
This code is naive.  There is not check that there is sufficient data in the tag to be processed by the chosen selector.